chore(translation): add Arabic localization

zaaakher opened the pull request:
<!-- Before creating a PR, please select one of the labels in the right-hand side Labels option: [feature], [fix], [documentation], [translation]. This will help Actions categorize PRs for Releases. -->

**By submitting this pull request, I confirm the following (please check the boxes):**

- [x] I have read and understood the [Contributor Guidelines](https://github.com/Redot-Engine/redot-website/blob/master/CONTRIBUTING.md).
- [x] I have verified that there are no duplicate pull requests for this request.
- [x] I believe this submission is valuable to others.
- [x] I acknowledge that this submission may not be used and the pull request may be closed at the maintainers' discretion.

**PR Content:**

- Adds the Arabic language

zaaakher commented on locales/ar/translation.json:
```diff
@@ -0,0 +1,321 @@
+      "heading": "محرك ألعاب،<br />صانع التغيير",
+      "description": "قم بإنشاء ألعابك ثنائية وثلاثية الأبعاد، ومشاريع متعددة المنصات، أو استكشف أفكارًا مبتكرة في تقنية XR مع محرك ريدوت!",
+      "buttons": {
+        "primary": "احصل على محرك ريدوت لنظام {os}",
```
This only looks like this in the code only. It's due to how the unicode of AR and EN are parsed different so it looks like it's off but if you run the website in localhost you'll see that it looks fine.

zaaakher commented on locales/ar/translation.json:
```diff
@@ -0,0 +1,321 @@
+        "noResults": "لم يتم العثور على لغة."
+      },
+      "copyright": {
+        "text": "© 2024-الحاضر بواسطة مساهمي مجتمع ريدوت.",
```
"Present" isn't really a technical term. However, should I use "present" in english? or should we use `{{date}}` and pass in the current year dynamically?

zaaakher commented:
I couldn't label the PR during the creation of the PR (the right side panel doesn't show up) and I can't add a label after submitting the PR due to access limitation.

charlottewiltshire0 commented:
@zaaakher I'm not an expert in linguistics, but as far as I know, Redot Engine is a proper name and it shouldn't be translated. You also forgot to add the flag of the country from the website: https://flagpack.xyz/, and you will need to select the large size there and place it in `public/flags`

zaaakher commented:
@charlottewiltshire0 Done 👍 

- I added the `SA.svg` flag from [here](https://github.com/Yummygum/flagpack-core/blob/main/svg/l/SA.svg?short_path=f37c89b)
- Corrected Redot name to be in English
- Changed the AR key in `/constants/langauges.ts` to `SA` in order to match with the flag name `SA.svg`

charlottewiltshire0 reviewed (approved):
Thank you so much for your contribution to the project. Are you a member of our Discord server?

zaaakher commented:
> Thank you so much for your contribution to the project. Are you a member of our Discord server?

Not yet. I actually don't have any game dev experience (yet) but I support the mission behind Redot 🌹 I will make sure to join the discord soon.

charlottewiltshire0 commented:
> sion behind Redot 🌹 I will make sure to join the discord soon.

If you join our Discord, let me or another staff member know so we can give you access to the redot-chan-lounge channel

zaaakher commented:
> If you join our Discord, let me or another staff member know so we can give you access to the redot-chan-lounge channel

Will do, thanks 🙏
